app/controllers/downloads_controller.rb

```diff
@@ -9,7 +9,10 @@ def latest
   end
 
   def guis
-    render "downloads/guis/index"
+    yaml = YAML.load_file('resources/guis.yml')
```

Comment:
I don't know much about the runtime environment for rails. Will we always be able to find the source file at this relative path?

Reply:
To be able to run the script, the present cwd had to be at the root of the project. Anyway, I don't see any useful case where we should make it run automatically.

Reply:
I agree that we don't need to worry much about the script. But this is running in the downloads controller of the rails app to render the GUI page. So it will be running on heroku for every request for that page.

Reply:
don't know a lot about the runtime environment for rails. it worked on localhost, not sure how it would behave on production. but running for every request may be a problem yes...

Reply:
Fixed this by having a singleton with the yaml contents so that the file is not read everytime but just on initialization. Not sure if it's the idiomatic way to do in Ruby/Rails. Let me know if there are better solutions @peff

Reply:
I'm not sure what's idiomatic. A singleton means we'd never notice if the file is updated. I think Rails has some magic that will detect changed on-disk files, but I'm not sure how to invoke it. It probably doesn't matter much for our purposes, though. In production we'd restart the process anytime there's an update anyway. So the only place it might bite is in a dev environment (e.g., you run `script/rails server` and then update the yaml, but the change isn't reflected).

Reply:
in production there's no problem, but in localhost changes may not be reflected yes. 

we can hack using `Rails.env.production?` for instance. In production or we want to avoid accessing the file for every request, or we want to check the file every time. Unless we decide for a more sophisticated solution, but I would say it's an overkill for our use case..

what do you guys think?

Reply:
@peff just pushed a new commit that uses `Rails.env.production?` to check if it should re-read the gui clients info at every request (the idea/hack is to ease developers doing changes on that part). I do think it's a bit hacky and dangerous, but I think it works for our use case. If you disagree we can revert the commit and try to find a solution (any other idea?).

Besides this issue, is anything else blocking merging this code PR that I can improve? :)